src/common/deployPipelineCache.ts

```diff
@@ -77,12 +77,13 @@ export class DeployPipelineCache extends TTLConfig<TTLConfig.Options, AsyncOpera
   }
 
   /**
-   * Returns the latest data cached
+   * Returns the latest data cached or throws if can't find any.
    */
-  public resolveLatest(): string {
+  public getLatestKeyOrThrow(): string {
     const aorId = this.getLatestKey();
-    if (!aorId) throw cacheMessages.createError('error.NoRecentAorId');
-
+    if (!aorId || aorId === 'undefined') {
```

Comment:
This is weird, the getLatestKey() method returns the string literal 'undefined' when it doesn't find any key.

Reply:
@EstebanRomero84 
Using the branch for this PR I added this test case to `test/common/deployPipelineCache.test.ts` and the function does not return 'undefined' for a key miss, it returns undefined.

```  it('return undefined when trying to get latest key in a new cache', async () => {
    const cache = await DeployPipelineCache.create();
    const entry = cache.getLatestKey();
    expect(entry).to.not.be.ok;
  });
```

Reply:
You are right, I'll fix it.
Thanks for the review.